sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -483,6 +508,11 @@ private CassandraIO() {}
               .collect(Collectors.joining(","));
 
       List<BoundedSource<T>> sources = new ArrayList<>();
+      String selectFields = "*";
+      if (spec.selectFields() != null) {
+        selectFields = String.join(",", spec.selectFields().get());
```

Comment:
Will do.

Reply:
@iemejia , thanks for feedback - challenge with withQuery in JdbcIO is that it is expected to be full query, like `select a from b` or `select a from b where c='d'`.
In CassandraIO select is generated so it can be split based on key ranges so 'where' part is partially dynamic so where part is set via separate field and I want select fields also set via seperate field/list of fields. 
That's why external builder can be challenge to support every case, unless it gives placeholder for each of those (select fields, where fields). 
So I generally agree we need some more generic but I don't think I could generalize mentioned MongoDB (BEAM-6241) and Cassandra query builder because they are totally different but definately I could try to create such builder for Cassandra that will gather projection/select fields and filter